File: lib/creational/abstract_factory/app/models/chair/victorian_chair.dart
import 'package:gof_design_patterns/creational/abstract_factory/app/models/chair/interfaces/chair.dart';

class VictorianChair with Chair {
  @override
  bool hasLegs() {
    return true;
  }

  @override
  void sitOn() {
    print('Siting on in a victorian chair');
  }
}

File: creational/abstract_factory/concrete_models/victorianChair.go
package concretemodels

import am "github.com/Victor-Fiamoncini/gof_design_patterns/creational/abstract_factory/abstract_models"

// VictorianChair struct model
type VictorianChair struct {
	Chair am.Chair
}

// SetMaterial VictorianChair method
func (vc *VictorianChair) SetMaterial(material string) {
	vc.Chair.Material = material
}

// SetHasLegs VictorianChair method
func (vc *VictorianChair) SetHasLegs(hasLegs bool) {
	vc.Chair.HasLegs = hasLegs
}

// GetMaterial VictorianChair method
func (vc *VictorianChair) GetMaterial() string {
	return vc.Chair.Material
}

// GetHasLegs VictorianChair method
func (vc *VictorianChair) GetHasLegs() bool {
	return vc.Chair.HasLegs
}
